ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/searchtree/algorithms/Leaf.java

```diff
@@ -193,10 +208,19 @@ void optimize(SearchTreeInput searchTreeInput, SearchTreeParameters parameters)
         if (status.equals(Status.EVALUATED) || status.equals(Status.OPTIMIZED)) {
             TECHNICAL_LOGS.debug("Optimizing leaf...");
 
+            // make a deep copy and change availableRangeAction
+            OptimizationPerimeter optimizationPerimeterWithFilteredHvdcRangeAction = searchTreeInput.getOptimizationPerimeter().copyWithFilteredAvailableRangeAction(network);
+
+            // check if there are still range actions to optimize
+            if (optimizationPerimeterWithFilteredHvdcRangeAction.getRangeActions().isEmpty()) {
+                TECHNICAL_LOGS.info("No range actions to optimize after filtering HVDC range actions");
```

Comment:
I think you should change the status to OPTIMIZED (en cohérence avec le commentaire de def de la méthode)